engine/runtime/src/main/java/org/enso/interpreter/runtime/callable/atom/AtomConstructor.java

```diff
@@ -43,6 +44,8 @@ public final class AtomConstructor implements TruffleObject {
   private @CompilerDirectives.CompilationFinal Atom cachedInstance;
   private @CompilerDirectives.CompilationFinal Function constructorFunction;
 
+  private final Type type;
```

Comment:
Is  there 1:1 mapping between `AtomConstructor` and `Type`, right now? E.g. it is more a refactoring and moving type-related code out of `AtomConstructor`.

How will the relation ship between constructor and type look like for `Vector<Boolean>` and `Vector<Text>`? Will the constructors be `==`? Now? In the future?